### PR TITLE
Fix cookies sending by ResponseEmitter.

### PR DESCRIPTION
**Pull request by ADmad**
Before this fix the cookies info in `Response::$_cookies` was never used to actually set cookies.

I haven't added/updated tests as I am not 100% sure this is the correct way to fix this problem but testing in an actual app it does seem to fix the problem.

### CODE REVIEW

File: src/Http/ResponseEmitter.php [hunk 1]
```diff
@@ -162,9 +162,11 @@ protected function emitStatusLine(ResponseInterface $response)
      */
     protected function emitHeaders(ResponseInterface $response)
     {
+        $cookies = $response->cookie();
```
Comment (antograssiot) [inline src/Http/ResponseEmitter.php]: Shouldn't it be `getCookie()` here ?

Comment (ADmad) [inline src/Http/ResponseEmitter.php]: Nope because `getCookie()` requires cookie name and doesn't return all cookies.

Comment (antograssiot) [inline src/Http/ResponseEmitter.php]: Then we should maybe check the response class before calling `cookie()` method to avoid the errors shown by the CI

**Comment by antograssiot** [comment]
solved it in my app also

File: src/Http/ResponseEmitter.php [hunk 2]
```diff
@@ -162,9 +162,14 @@ protected function emitStatusLine(ResponseInterface $response)
      */
     protected function emitHeaders(ResponseInterface $response)
     {
+        $cookies = [];
+        if (method_exists($response, 'cookie')) {
+            $cookies = $response->cookie();
```
Comment (markstory) [inline src/Http/ResponseEmitter.php]: I wonder if we should add a getCookies() method.

**Review by markstory — COMMENTED** [review]
I can add  a test before merging this.

File: src/Http/ResponseEmitter.php (hunk 2 above)
Comment (markstory) [inline src/Http/ResponseEmitter.php]: I added the `getCookies()` method and added tests.

**Comment by antograssiot** [comment]
`Cake\Test\TestCase\Network\ResponseTest::testGetReasonPhrase`also needs to be updated if the default phrase is `'OK'` instead of null

**Comment by markstory** [comment]
I will get that fixed.

**Pull request merged**